source/simulator/assembly.cc

```diff
@@ -578,14 +578,20 @@ namespace aspect
         const Tensor<1,dim> velocity = (scratch.old_velocity_values[q] +
                                         scratch.old_old_velocity_values[q]) / 2;
 
+        const SymmetricTensor<2,dim> strain_rate_tensor = (scratch.old_strain_rates[q] +
+                                                           scratch.old_old_strain_rates[q]) / 2;
+        const double strain_rate = std::sqrt(strain_rate_tensor * strain_rate_tensor);
```

Comment:
I think this is `strain_rate_tensor.norm()`. You may not even need to name the strain rate tensor object if you just write
```
  const double ... = ((old_s_r[q] + o_o_strain_r[q])/2).norm();
```